src/p2p.rs

```diff
@@ -350,14 +363,17 @@ struct RawNetworkMessage {
 }
 
 impl RawNetworkMessage {
-    fn parse(&self) -> Result<NetworkMessage> {
+    fn parse(self) -> Result<NetworkMessage> {
         let mut raw: &[u8] = &self.raw;
         let payload = match self.cmd.as_ref() {
             "version" => NetworkMessage::Version(Decodable::consensus_decode(&mut raw)?),
             "verack" => NetworkMessage::Verack,
             "inv" => NetworkMessage::Inv(Decodable::consensus_decode(&mut raw)?),
             "notfound" => NetworkMessage::NotFound(Decodable::consensus_decode(&mut raw)?),
-            "block" => NetworkMessage::Block(Decodable::consensus_decode(&mut raw)?),
+            "block" => NetworkMessage::Unknown {
+                command: self.cmd.clone(),
```

Comment:
`clone` is not needed here, can simply take ownership of `self.cmd`

Reply:
thanks, fixed in e80a96365e0b6cd3bc461359327bd9c1fb729495